genesyscloud/consistency_checker/consistency_checker.go

```diff
@@ -0,0 +1,285 @@
+	if len(unfilteredSlice) == 0 {
+		return unfilteredSlice
+	}
+
+	switch unfilteredSlice[0].(type) {
+	case map[string]interface{}:
+		filteredSlice := make([]interface{}, 0)
+		for _, s := range unfilteredSlice {
+			filteredSlice = append(filteredSlice, filterMap(s.(map[string]interface{})))
+		}
+		return filteredSlice
+	}
+
+	return unfilteredSlice
+}
+
+func filterMap(m map[string]interface{}) map[string]interface{} {
+	newM := make(map[string]interface{})
+	for k, v := range m {
+		switch t := v.(type) {
+		case *schema.Set:
+			newM[k] = filterMapSlice(t.List())
+		case []interface{}:
+			newM[k] = filterMapSlice(t)
+		case map[string]interface{}:
+			if len(t) > 0 {
+				newM[k] = filterMap(t)
+			}
+		default:
+			newM[k] = v
+		}
+	}
+
+	return newM
+}
+
+func compareValues(oldValue, newValue interface{}, index, index2 int, key string) bool {
+	switch oldValueType := oldValue.(type) {
+	case []interface{}:
+		if index >= len(oldValueType) {
+			return false
+		}
+		ov := oldValueType[index]
+		switch t := ov.(type) {
+		case map[string]interface{}:
+			return compareValues(t[key], newValue, index2, 0, "")
+		default:
+			return cmp.Equal(ov, newValue)
+		}
+	case string:
+		return cmp.Equal(oldValue, newValue)
+	}
+
+	return true
+}
+
+func isNilOrEmpty(value interface{}) bool {
+	if value == nil {
+		return true
+	}
+
+	switch valueType := value.(type) {
+	case int:
+		return valueType == 0
+	case string:
+		return valueType == ""
+	case []interface{}:
+		return len(valueType) == 0
+	case *schema.Set:
+		if valueType == nil {
+			return true
+		}
+		return valueType.Len() == 0
+	default:
+		// An interface holding a nil value is not nil. Need to use reflection to see if it's nil
+		kind := reflect.ValueOf(value).Kind()
+		if kind == reflect.Map {
+			return reflect.ValueOf(value).IsNil()
+		}
+	}
+
+	return false
+}
+
+func (c *consistencyCheck) isComputed(key string) bool {
+	schemaInterface := getUnexportedField(reflect.ValueOf(c.d).Elem().FieldByName("schema"))
+	resourceSchema := schemaInterface.(map[string]*schema.Schema)
+
+	k := key
+	if strings.Contains(key, ".") {
+		k = strings.Split(key, ".")[0]
+	}
+	if resourceSchema[k] == nil {
+		return false
+	}
+
+	return resourceSchema[k].Computed
+}
+
+func (c *consistencyCheck) CheckState() *resource.RetryError {
+	if c.isEmptyState == nil {
+		panic("consistencyCheck must be initialized with NewConsistencyCheck")
+	}
+
+	if *c.isEmptyState {
+		return nil
+	}
+
+	originalState := filterMap(c.originalState)
+	resourceConfig := &terraform.ResourceConfig{
+		ComputedKeys: []string{},
+		Config:       originalState,
+		Raw:          originalState,
+	}
+	diff, _ := c.r.SimpleDiff(c.ctx, c.d.State(), resourceConfig, c.meta)
+	if diff != nil {
+		//fmt.Println(diff)
+		for k, v := range diff.Attributes {
+			if strings.HasSuffix(k, "#") {
+				continue
+			}
+
+			//fmt.Println("!c.d.HasChange(k)", !c.d.HasChange(k))
+			//fmt.Println("isNilOrEmpty(c.d.Get(k))", isNilOrEmpty(c.d.Get(k)))
```

Comment:
We should remove this commented code.  If its is logging information we should use the Go log function